### ISSUE TITLE
[Question] Is it possible to fetch filtered collections with joins?

### ISSUE THREAD

**Issue by loalf** [doctrine/doctrine2#6300]
Hi,

I've spent the last couple of weeks trying to figure out whether what I'm trying to achieve is either impossible or just plain silly.

Here is a simplified version of my use case. Let's say we have users, posts and post_statuses (draft, public, private, ...). I'd like to fetch all active posts for a given users. Ideally, I'd like to have this API:

```php
$user->getActivePosts()
```

I found out a really nice feature in Doctrine2, `matching` on `Collections`, the problem is that it doesn't allow joins in it.

Yes, I know, for this specific example I could avoid having a join if I were to do something like

```php
class User {
  protected $posts;

  public function getActivePosts() {
    $criteria = Criteria::create()->where(Criteria::expr()->eq("status_id", 1))
    return $this->posts->matching($criteria);
  }
}
```

but bare with me and let's assume that I do have to join a third table.

Obviously, I'm aware that I could do something like this and move this logic to the repository class, but it just seems wrong to me.

```php
$entityManager->getRepository('Post')->findActivePostsByUser($user)
```

This has been bugging me for the last two weeks so even a quick answer like "sorry, can't do that" would be a massive relief.

Thanks,
Javi

**Comment by Ocramius** [comment]
I assume that `status_id` should instead be `status` and a target entity instance?

**Comment by loalf** [comment]
@Ocramius that's correct. BTW, you are the main reason why this is bugging me, 😉 . I watched this talked recently, https://github.com/doctrine/doctrine2/issues/6300#issuecomment-282082139, and now I can get the law of Demeter out of my head, 😝  (really good talk, kudos!)

**Comment by loalf** [comment]
@Ocramius shall I cease and desist? ;)

**Comment by Ocramius** [comment]
@loalf I'm not sure if this is supposed to work when fetching by the actual referenced status entity. Could give it a shot, but I don't think it will work.

**Comment by holtkamp** [comment]
>Obviously, I'm aware that I could do something like this and move this logic to the repository class, but it just seems wrong to me.

Encountered this as well, I accepted that in some cases, the Model (Entity) needs to relay an incoming request to its Repository. To me this seems "better" / more technology agnostic than having this in your Model:

```php
Criteria::create()->where(Criteria::expr()->eq("status", 1))
```
